lib/k8s/conn/auth/exec.ex

```diff
@@ -31,42 +31,45 @@ defmodule K8s.Conn.Auth.Exec do
   """
 
   @behaviour K8s.Conn.Auth
+
   alias __MODULE__
+  alias K8s.Conn.Auth.ExecWorker
   alias K8s.Conn.Error
 
-  defstruct [:command, :env, args: []]
+  defstruct [:target]
 
-  @type t :: %__MODULE__{
-          command: String.t(),
-          env: %{name: String.t(), value: String.t()},
-          args: list(String.t())
-        }
+  @type t :: %__MODULE__{target: GenServer.server()}
 
   @impl true
   @spec create(map() | any, String.t() | any) :: {:ok, t} | {:error, Error.t()} | :skip
-  def create(%{"exec" => %{"command" => command} = config}, _) do
-    # Optional:
-    args = config["args"] |> List.wrap()
-    env = config["env"] |> List.wrap() |> format_env()
-
-    {:ok,
-     %__MODULE__{
-       command: command,
-       env: env,
-       args: args
-     }}
+  def create(%{"exec" => %{}} = ctx, _) do
+    # We don't want to be in the habit of parsing command names for unique identifiers
+    # so every auth worker gets it's own unique name. Then we use that name to
+    # register the worker with the registry.
+    id = :crypto.strong_rand_bytes(16) |> Base.url_encode64(padding: false)
```

Comment:
The registry and naming allow for GenServers to crash and still have a working `Conn`.

Imagine this sequence.

- User creates a new K8s.Conn
- The Exec worker starts with a pid of 12 since it's the 12th process.
- `K8s.Conn` needs a token so it sends a request to pid 12
- Something strange happens with calling the exec binary (not sure what, but something)
- The exec worker gets restarted with a pid of 13 by the supervisor
- The `K8s.Conn` needs a token so it sends pid 12 a request.
- The `K8s.Conn` is broken and won't work again.

With the name and the registry we get a way to look that up. So the sequence with the current strucutre would be:


- User creates a new K8s.Conn
- The Exec worker starts with a pid of 12 since it's the 12th process.
- The registry gets a mapping of name to pid.
- `K8s.Conn` needs a token
- We ask registry for a pid to send a request
- We send the request to pid 12
- Something strange happens with calling the exec binary (not sure what, but something)
- The exec worker gets restarted with a pid of 13 by the supervisor
- The registry gets the new mapping of name to pid 13
- The `K8s.Conn` needs a token so it  checks the registry and gets pid 13
- The `K8s.Conn` works after the process restarts.

Reply:
Yes you are right of course. Thanks for the detailed explanations.